packages/cosmo-eccodes-definitions/package.py

```diff
@@ -43,16 +43,7 @@ class CosmoEccodesDefinitions(Package):
     version('2.14.1.2', tag='v2.14.1.2')
     version('2.14.1.1', tag='v2.14.1.1')
 
-    depends_on('eccodes')
-    depends_on('eccodes@2.19.0',
-               type=('build', 'link', 'run'),
-               when='@2.19.0.1:')
-    depends_on('eccodes@2.18.0',
-               type=('build', 'link', 'run'),
-               when='@2.18.0.1')
-    depends_on('eccodes@2.14.1',
-               type=('build', 'link', 'run'),
-               when='@2.14.1.1:2.14.1.2')
+    depends_on('eccodes +fortran')
```

Comment:
Then change is required because we replaced our eccodes package with the built-in of Spack. There the Fortran-Interface is disabled by default.
And yes COSMO needs the Fortran-Interface:
https://github.com/C2SM/spack-c2sm/blob/8f999cb8aaa6cfb98b8b3f0e7b7a13cbf699429c/packages/cosmo/package.py#L240

Reply:
Then COSMO needs to depend on `cosmo-eccodes-definitions ^eccodes +fortran`, not cosmo-eccodes-definitions.

Reply:
I find it a bit counterintuitive that COSMO depends on eccodes  via cosmo-eccodes-definitions.
Also because we explicitely link against eccodes-libs in the cosmo-makefile. cosmo-eccodes-definitions itself only set some runtime-paths.

My suggestion:

- Remove eccodes as dependency of cosmo-eccodes-definitions
- Properly introduce `eccodes + fortran` as a dep of cosmo package itself.

Picking the right combination of eccodes and cosmo-eccodes-definitons is then entirley outsourced to spec.yaml. Were it should really be.

Reply:
You are absolutely right, that's weird.
So yes on introducing `depends_on('eccodes +fortran', ...)` in COSMO.

Reply:
I made eccodes a direct dep of cosmo now.
Unfortunately I cannot remove eccodes from the deps of the cosmo-ecccodes-definitions package because it set some paths using eccodes.

Reply:
I think this is okay. As far as I understand eccodes/cosmo-ecccodes-definitions is a database. Eccodes containing some executable code and some data, and cosmo-ecccodes-definitions only containing data. So the data per se doesn't depend on any thing, it's just bits. But in order to be able to interpret the bits one depends on a specific version of eccodes which defines how the data has to be interpreted. Ideally the data should be spack-resources in eccodes but I think this is an issues for an other time as atm Sergey is working on that: https://github.com/spack/spack/pull/29299